File: core/src/main/java/org/ravine/core/server/cluster/ClusterUtils.java
package org.ravine.core.server.cluster;

import org.ravine.core.config.Config;
import org.ravine.shared.schemes.URI;

public class ClusterUtils {
  private ClusterUtils() { }

  public static Node buildSelfNode(final Config config) {
    final URI uri = URI.parseURIString(config.get(Config.KEY_HOST) + ":" + config.getAsInteger(Config.KEY_PORT));
    return new Node(uri);
  }
}

File: core/src/main/java/org/ravine/core/server/cluster/ClusterUtils.kt
package org.ravine.core.server.cluster

import org.ravine.core.Environment
import org.ravine.core.config.Config
import org.ravine.shared.schemes.URI

object ClusterUtils {
  fun buildSelfNode(): Node {
    val host: String = Environment.config[Config.KEY_HOST]
    val port: Int = Environment.config[Config.KEY_PORT]
    return Node(URI.parseURIString("$host:$port"))
  }
}
